### ISSUE TITLE
error: link modifiers combination `+bundle,+whole-archive` is unstable when generating rlibs

### ISSUE THREAD

**Issue by CarlSchwan** [KDAB/cxx-qt#519]
```
Compiling qml-features v0.1.0 (/home/cschwan/kde/cxx-qt/examples/qml_features/rust)
error: link modifiers combination `+bundle,+whole-archive` is unstable when generating rlibs
```

tested with rust from the fedora repo and from rustup.

**Comment by ahayzen-kdab** [comment]
Manually setting `cc_builder_whole_archive.link_lib_modifier("-bundle");` to still allow `+whole-archive` solves the issue.

But we need to check that

- Is it right that a `rlib` is being built there?
- Is setting `-bundle` OK or does that cause other problems?

@Be-ing ^^ any opinions ?

**Comment by Be-ing** [comment]
https://github.com/rust-lang/rust/issues/108081 :thinking:

**Comment by Be-ing** [comment]
Upstream fix: https://github.com/rust-lang/rust/pull/110917

**Comment by Be-ing** [comment]
Fix was merged upstream and will be in Rust 1.71 and maybe 1.70.

**Comment by ahayzen-kdab** [comment]
@Be-ing do we want to leave this open as a reminder to remove the workaround ? And also do we want to remove the workaround at some point ? How long should we wait for people to move on from 1.69 to 1.70+ ? Or is it even possible to have a cfg flag for a specific rustc version ?

**Comment by Be-ing** [comment]
The workaround is harmless AFAIK, so we might as well leave it forever.

**Comment by ahayzen-kdab** [comment]
Cool, just wanted to check :-)

**Comment by Be-ing** [comment]
We could remove the workaround if we raise the MSRV to >= 1.70 in the future (or 1.71 if the fix doesn't make it into 1.70).

**Comment by Be-ing** [comment]
https://github.com/rust-lang/rust/pull/110917 was approved for backporting to 1.70.
